src/music.c

```diff
@@ -56,6 +56,7 @@ char *music_cmd = NULL;
 static SDL_bool music_active = SDL_TRUE;
 static int music_volume = MIX_MAX_VOLUME;
 static Mix_Music * volatile music_playing = NULL;
+static Uint8 *mix_music_buffer = NULL;
```

Comment:
Is this one safe?

I understand that one can have multiple music instances loaded by Mix_LoadMUS,
but there is only one `music_playing` pointing only to one. I want to be sure though.

Reply:
This music buffer is used directly at the playing side only and has never been used in other cases. `music_mixer()` call is only one user here that works in one thread. 

Trivia: On the MixerX side, I have a similar buffer, but I use it to mix multiple music streams playing in parallel. Single-music player I kept being the same, and the multi-music player I made as a separate addition that works in parallel.

Reply:
I don't like it though. That buffer can become a member of `struct _Mix_Music`
and as such become per-music, can it not?

Reply:
@slouken: What's your overall opinion, and your opinion about using that secondary buffer for music + master volume?

Reply:
>  and as such become per-music, can it not?

The idea of this buffer is the ability to separately mix multiple playing music streams (which I did at the MixerX side where I use this buffer to take a chunk of playing music with all applied effects and fade, then safely mix it into the output stream). You may review the implementation here: https://github.com/WohlSoft/SDL-Mixer-X/blob/master/src/music.c#L760-L858

Reply:
We don't have a feature of playing and mixing multiple music streams though

Reply:
> We don't have a feature of playing and mixing multiple music streams though


I do think to try to submit it to mainstream Mixer once upon. This has many ways to use it:
- You can play heavy atmosphere environment sounds that aren't suitable for chunks API because of long loading and big memory usage.
- Allows running multiple parallel tracks of the same song and mute/unmuting each of them by necessary.
- Allows cross-fade of music from one to another. (by current music API it's basically impossible).

And, to keep the backward compatibility, I made a new set of functions that keeps the original API work as-is.

Anyway, as was discussed a while ago, some music engines won't work on multi-music API (such as MikMod and Native MIDI) as they were able to play only one song at the moment. So, in an attempt to load the thing using the multi-music API, the error will be returned.